scope_shared/scope/database/patient/activity_schedules.py

```diff
@@ -148,6 +148,52 @@ def _maintain_pending_scheduled_activities(
             )
 
 
+def delete_activity_schedule(
+    *,
+    collection: pymongo.collection.Collection,
+    set_id: str,
+    rev: int,
+) -> scope.database.collection_utils.SetPutResult:
+    """
+    Delete "activity-schedule" document.
+
+    - Any corresponding ScheduledActivity documents must be deleted.
+    """
+
+    result = scope.database.collection_utils.delete_set_element(
+        collection=collection,
+        document_type=DOCUMENT_TYPE,
+        set_id=set_id,
+        rev=rev,
+    )
+
+    if result.inserted_count == 1:
+        existing_scheduled_activities = (
+            scope.database.patient.scheduled_activities.get_scheduled_activities(
+                collection=collection
+            )
+        )
+
+        # Perform only the delete component of maintenance
+        delete_items = _calculate_scheduled_activities_to_delete(
```

Comment:
@jayfo Not completely sure if the test in `test_delete_activity_schedule...` is covering all the edge cases?

Reply:
I agree, but those same code paths are more explicitly tested in `test_activity_schedule_maintains_scheduled_activities.py`. Since I'm re-using them, this felt fine.